File: backend/src/main/java/com/domain80/wholistika/features/userAccount/dto/UserAccountDto.java
package com.domain80.wholistika.features.userAccount.dto;

import com.domain80.wholistika.features.userAccount.models.MedicalCondition;
import com.domain80.wholistika.features.userAccount.models.WorkExperience;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class UserAccountDto {

    private String id;
    private String firstName;
    private String lastName;
    private String email;
    private String phoneNumber;
    private String role;
    private Boolean isNewUser;

    private String title;
    private String jobTitle;
    private String interests;
    private String about;
    private String imageData;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private List<MedicalCondition> medicalConditions;
    private List<WorkExperience> workExperiences;
}

File: frontend/src/features/onboarding/auth/dto/userAccount.dto.ts
export class UserAccountDto {
  id!: string
  email!: string
  firstName!: string
  lastName!: string
  title?: string
  jobTitle?: string
  interests?: string
  about?: string
  imageData?: string
  createdAt?: Date
  updatedAt?: Date

  medicalInfo?: Array<{
    name: string
    summary: string
  }>
  workExperiences?: Array<{
    title: string
    where: string
    startDate: Date
    endDate?: Date
    currentlyWork: boolean
    jobSummary: string
  }>
}
